### PR TITLE
feature/AT-SCE-UI_InitialStructure

### PR DESCRIPTION
**Pull request by JuanGalvez12**
Added the index.js file and config files.

### CODE REVIEW

File: src/routes/admin.routes.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+// AT-SCE - Admin - Users - Add User
+router.post("/user/add", addUser);
+
+// AT-SCE - Admin - Users - Render Edit User Form
+router.get("/user/edit/:id", renderEditUserForm);
+// AT-SCE - Admin - Users - Edit User
+router.put("/user/edit/:id", updateUser);
+
+// AT-SCE - Admin - Users - Delete User
+router.get("/user/delete/:id", deleteUser);
```
Comment (luishcastroc) [inline src/routes/admin.routes.js:33]: this should be a delete per standard?

File: src/routes/at-sce.routes.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+// AT-SCE - Index
+router.get("/", renderSigninForm);
+
+// AT-SCE - Login
+router.get("/signin", renderSigninForm);
+router.post("/signin", signin);
+
+// AT-SCE - Logout
+router.get("/signout", signout);
```
Comment (luishcastroc) [inline src/routes/at-sce.routes.js:23]: signout should be a post

File: src/server.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+const session = require("express-session");
+const flash = require("connect-flash");
+
+// Initializations
+const app = express();
+
+// Settings
+app.set("port", process.env.PORT || 4000);
+/*app.set("views", path.join(__dirname, "views"));
+app.engine(
```
Comment (luishcastroc) [inline src/server.js:16]: is all of this commented code something that will be used? otherwise don't see the point on leaving it here.

**Review by luishcastroc — CHANGES_REQUESTED** [review]

File: package.json [hunk 4]
```diff
@@ -16,5 +17,23 @@
...
+    "express-session": "^1.17.1",
+    "handlebars": "^4.7.7",
+    "method-override": "^3.0.0",
+    "mongoose": "^5.11.18",
+    "morgan": "^1.10.0",
+    "nodemon": "^2.0.7",
+    "npm-check-updates": "^11.1.9",
+    "passport": "^0.4.1",
+    "passport-local": "^1.0.0"
+  }
```
Comment (at-internship) [inline package.json:49]: Please add these dependencies:
"devDependencies": {
    "dotenv": "^8.2.0",
    "handlebars": "^4.7.6",
    "nodemon": "^2.0.7",
    "npm-check-updates": "^9.2.4"
  },

File: src/controllers/admin.controller.js [hunk 5]
```diff
@@ -1,22 +1,24 @@
...
+
+adminCtrl.renderIndex = async(req, res) =>{
 };
 
-adminCtrl.renderUserList = async () => {
+adminCtrl.renderUserList = async(req, res) =>{
 }; 
 
-adminCtrl.renderAddUserForm = async () => {
+adminCtrl.renderAddUserForm =async(req, res) =>{
```
Comment (at-internship) [inline src/controllers/admin.controller.js]: Take care with this kind of changes (blank spaces), please be sure you have these plugins:
Integrated Terminal 
Prettier 
Material Icon Theme

File: src/index.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+if (process.env.NODE_ENV !== "production") {
+    require("dotenv").config();
+  }
```
Comment (at-internship) [inline src/index.js]: Take care with this kind of changes (blank spaces), please be sure you have these plugins:
Integrated Terminal
Prettier
Material Icon Theme

File: src/routes/at-sce.routes.js (hunk 2 above)
Comment (at-internship) [reply src/routes/at-sce.routes.js:23]: I think the "GET" path is fine here (it does not refer to the HTTP operation)

File: src/routes/admin.routes.js (hunk 1 above)
Comment (at-internship) [reply src/routes/admin.routes.js:33]: I think the "GET" path is fine here (it does not refer to the HTTP operation)

File: src/server.js [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+  ".hbs",
+  exphbs({
+    defaultLayout: "main",
+    layoutsDir: path.join(app.get("views"), "layouts"),
+    partialsDir: path.join(app.get("views"), "partials"),
+    extname: ".hbs",
+
+    // Helpers
```
Comment (at-internship) [inline src/server.js:24]: Please remove this helper section

**Review by at-internship — CHANGES_REQUESTED** [review]
Please verify

**Review by at-internship — APPROVED** [review]

File: src/routes/admin.routes.js (hunk 1 above)
Comment (AT-ivillamil) [reply src/routes/admin.routes.js:33]: Agree, this is route for an app view, not an API request

File: src/controllers/admin.controller.js [hunk 8]
```diff
@@ -1,22 +1,179 @@
...
 
-adminCtrl.renderEditUserForm = async () => {
+// AT-SCE - Admin - Users - Add User
+adminCtrl.addUser = async (req, res) => {
+  console.log("--> adminCtrl.addUser");
+
+  try {
+    const {
+      user_name,
+      user_firstName,
```
Comment (luishcastroc) [inline src/controllers/admin.controller.js:55]: you are justin user_name and then you switch to user_firstName, use a Case convention either kebab or camelCase, i see that the names on the controllers are in camelCase, why not to use that to be consistent?

File: src/controllers/admin.controller.js [hunk 9]
```diff
@@ -1,22 +1,179 @@
...
+    } = req.body;
+
+    //const userErrors = [];
+
+    let users;
+
+    let request = {
+      name: user_name,
+      firstName: user_firstName,
+      lastName: user_lastName,
```
Comment (luishcastroc) [inline src/controllers/admin.controller.js:69]: now in here we use camelCase, we need some consistency between back and front end if this are the properties the backend is sending then we need to set a meeting with them to agree on a case or check documentation of the project to see if that is mentioned.

File: package.json [hunk 10]
```diff
@@ -1,20 +1,50 @@
...
+    "express-session": "^1.17.1",
+    "method-override": "^3.0.0",
+    "mocha": "^8.2.1",
+    "morgan": "^1.10.0",
+    "nock": "^13.0.5",
+    "nyc": "^15.1.0",
+    "passport": "^0.4.1",
+    "passport-local": "^1.0.0",
+    "sinon": "^9.2.1"
+  },
```
Comment (luishcastroc) [inline package.json:40]: add prettier to have a single source of formatting

**Review by irvingesp2190 — APPROVED** [review]

File: src/controllers/admin.controller.js [hunk 11]
```diff
@@ -1,22 +1,183 @@
...
+ *
+ * General functions for at-admin-controller.
+ *
+<<<<<<< HEAD
+ *
+=======
+ * @author @at-internship
+>>>>>>> feature/AT-SCE_POC
+ * @version 1.0
+ */
```
Comment (AT-ivillamil) [inline src/controllers/admin.controller.js]: This is a merge conflict

File: src/controllers/at-sce.controller.js [hunk 12]
```diff
@@ -1,15 +1,49 @@
...
+ *
+ * General functions for at-sce-controller.
+ *
+<<<<<<< HEAD
+ *
+=======
+ * @author @at-internship
+>>>>>>> feature/AT-SCE_POC
+ * @version 1.0
+ */
```
Comment (AT-ivillamil) [inline src/controllers/at-sce.controller.js]: Merge conflict

**Review by AT-ivillamil — CHANGES_REQUESTED** [review]
Admin views hasn't been created yet but they are referenced in `admin.controller.js`, I think we should just return a placeholder text on all admin routes for now, otherwise the app will show an error if we try to access any of the admin routes.

File: src/public/css/main.css [hunk 13]
```diff
@@ -0,0 +1,144 @@
...
+}
+
+.help_icon {
+  background-image: url(public/img/icons/help-icon.png);
+  width: 21px;
+}
+
+@media screen and (min-width: 320px) {
+  html {
+    font-size: calc(12px + 6 * ((100vw - 320px) / 680));
```
Comment (luishcastroc) [inline src/public/css/main.css:35]: this could be easier with clamp

**Review by AT-ivillamil — APPROVED** [review]

**Review by Emiliano-Plascencia — APPROVED** [review]

**Review by luishcastroc — APPROVED** [review]

**Pull request merged**